### PR TITLE
kuttl: init at version 0.9.0

### PR DESCRIPTION
**Pull request by DiegoLelis**
###### Motivation for this change
Tool to test k8s clusters declaratively

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2339,6 +2339,12 @@
     githubId = 15774340;
     name = "Thomas Depierre";
   };
+  diegolelis = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2342]: Please put this into a separate commit with the message ``maintainers: add diegolelis``.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -23702,6 +23702,11 @@ in
 
   kubectl = callPackage ../applications/networking/cluster/kubectl { };
 
+  kuttl = callPackage ../applications/networking/cluster/kuttl {
+    # This package does not work with Go 1.16
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:23706]:
```suggestion
```

File: pkgs/applications/networking/cluster/kuttl/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    -X  github.com/kudobuilder/kuttl/pkg/version.gitVersion=${version} 
+  '';
+
+  meta = with lib; {
+    description = "The KUbernetes Test TooL (KUTTL) provides a declarative approach to testing production-grade Kubernetes operators";
+    homepage = "https://github.com/kudobuilder/kuttl";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ diegolelis ];
+  };
+}
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/kuttl/default.nix:32]: Missing final new line.

File: pkgs/applications/networking/cluster/kuttl/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+
+{ lib, buildGoModule, fetchFromGitHub}:
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/kuttl/default.nix:3]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub}:

```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please change the package init commit to ``kuttl: init at version 0.9.0``.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (DiegoLelis) [reply maintainers/maintainer-list.nix:2342]: Done

File: pkgs/applications/networking/cluster/kuttl/default.nix (hunk 3 above)
Comment (DiegoLelis) [reply pkgs/applications/networking/cluster/kuttl/default.nix]: Done

File: pkgs/applications/networking/cluster/kuttl/default.nix (hunk 4 above)
Comment (DiegoLelis) [reply pkgs/applications/networking/cluster/kuttl/default.nix]: Done

**Comment by kalbasit** [comment]
@DiegoLelis you seem to have some whitespace issues there. Could you please amend your commits and remove the trailing whitespaces.

**Comment by DiegoLelis** [comment]
> @DiegoLelis you seem to have some whitespace issues there. Could you please amend your commits and remove the trailing whitespaces.

Done @kalbasit
